### ISSUE TITLE
attention mask

### ISSUE THREAD

**Issue by ZOUHAN1** [openai/CLIP#488]
I would like to understand why masking is used in the text encoder. This doesn't seem necessary for CLIP since it does not perform an autoregressive task. Maybe my understanding is incomplete. The relevant code is located at line 286 in model.py.

**Comment by wadkisson** [comment]
What up @ZOUHAN1. The paper released with the repo mentions on page five that "Masked self-attention was used in the text encoder to preserve the ability to initialize with a pre-trained language model or add language modeling as an auxiliary objective...".  From what I'm hearing, the authors implemented masked attention just in case they wanted to experiment with using pre-trained models down the line.
